File: v1/src/main/java/com/api/v1/medical_slots/CancelledMedicalSlotResponseDTO.java
package com.api.v1.medical_slots;

import com.api.v1.doctors.DoctorResponseDTO;
import com.api.v1.medical_slots.enums.MedicalSlotStatus;

import java.time.LocalDateTime;

public final class CancelledMedicalSlotResponseDTO extends MedicalSlotResponseDTO {

    private final LocalDateTime cancelledAt;

    private CancelledMedicalSlotResponseDTO(String id,
                                           MedicalSlotStatus status,
                                           DoctorResponseDTO doctor,
                                           LocalDateTime availableAt,
                                           LocalDateTime createdAt,
                                           LocalDateTime cancelledAt
    ) {
        super(id, status, doctor, availableAt, createdAt);
        this.cancelledAt = cancelledAt;
    }

    public static CancelledMedicalSlotResponseDTO from(MedicalSlot medicalSlot) {
        return new CancelledMedicalSlotResponseDTO(
                medicalSlot.getId(),
                MedicalSlotStatus.CANCELLED,
                medicalSlot.toDTO().getDoctor(),
                medicalSlot.getAvailableAt(),
                medicalSlot.getCreatedAt(),
                medicalSlot.getCancelledAt()
        );
    }

    public LocalDateTime getCancelledAt() {
        return cancelledAt;
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/CancelledMedicalSlotResponseDTO.kt
package com.api.v2.medical_slots

import com.api.v2.doctors.DoctorResponseDTO
import com.api.v2.medical_slots.enums.MedicalSlotStatus
import java.time.LocalDateTime

class CancelledMedicalSlotResponseDTO(
    id: String,
    doctor: DoctorResponseDTO,
    availableAt: LocalDateTime,
    createdAt: LocalDateTime,
    cancelledAt: LocalDateTime
) : MedicalSlotResponseDTO(id, MedicalSlotStatus.CANCELLED, doctor, availableAt, createdAt)